### PR TITLE
Linux sysdep additions and stubbing from LFS

### PR DESCRIPTION
**Pull request by no92**
Split out from #651.

### CODE REVIEW

File: options/posix/generic/grp-stubs.cpp [hunk 1]
```diff
@@ -236,13 +236,13 @@ int getgrnam_r(const char *name, struct group *grp, char *buffer, size_t size, s
...
 
-	*result = grp;		
+	*result = grp;
 	return 0;
 }
 
 void setgrent(void) {
-	__ensure(!"Not implemented");
-	__builtin_unreachable();
+	// TODO: once getgrent is implemented, this needs to rewind to the beginning of the database if necessary
```
Comment (64) [inline options/posix/generic/grp-stubs.cpp:244]: It'd be nice to print a warning here

File: options/posix/generic/sched-stubs.cpp [hunk 2]
```diff
@@ -49,3 +54,8 @@ int clone(int (*)(void *), void *, int, void *, ...) {
 	__ensure(!"Not implemented");
 	__builtin_unreachable();
 }
+
+int sched_getcpu(void) {
```
Comment (ArsenArsen) [inline options/posix/generic/sched-stubs.cpp:58]: stubbing this isn't necessary

File: options/posix/include/netdb.h [hunk 3]
```diff
@@ -105,6 +105,7 @@ int getaddrinfo(const char *__restrict, const char *__restrict,
 		const struct addrinfo *__restrict, struct addrinfo **__restrict);
 struct hostent *gethostent(void);
 struct hostent *gethostbyname(const char *);
+struct hostent *gethostbyname2(const char *, int);
```
Comment (ArsenArsen) [inline options/posix/include/netdb.h:108]: where's the implementation?

**Review by 64 — CHANGES_REQUESTED** [review]

**Review by ArsenArsen — CHANGES_REQUESTED** [review]

File: options/posix/include/netdb.h (hunk 3 above)
Comment (no92) [reply options/posix/include/netdb.h:108]: In `options/posix/generic/netdb-stubs.cpp`.

Comment (ArsenArsen) [reply options/posix/include/netdb.h:108]: oh, huh, I didn't see it

**Review by 64 — APPROVED** [review]

**Review by ArsenArsen — APPROVED** [review]

File: options/posix/generic/sched-stubs.cpp [hunk 4]
```diff
@@ -1,9 +1,10 @@
 
 #include <bits/ensure.h>
+#include <errno.h>
 #include <sched.h>
 
 #include <mlibc/debug.hpp>
-#include <mlibc/posix-sysdeps.hpp>
+#include <mlibc/all-sysdeps.hpp>
```
Comment (Geertiebear) [inline options/posix/generic/sched-stubs.cpp:7]: Why this change? The posix option should only include the posix sysdeps, if posix needs a sysdep not in the header then it should be added.

File: options/posix/generic/sched-stubs.cpp [hunk 5]
```diff
@@ -49,3 +54,17 @@ int clone(int (*)(void *), void *, int, void *, ...) {
 	__ensure(!"Not implemented");
 	__builtin_unreachable();
 }
+
+#ifdef __MLIBC_LINUX_OPTION
```
Comment (Geertiebear) [inline options/posix/generic/sched-stubs.cpp:58]: This is probably better handled with an extra ``linux-sched`` file in the linux option.

**Review by Geertiebear — CHANGES_REQUESTED** [review]

**Comment by no92** [comment]
Rebased onto master and addressed the comments raised.

**Review by ArsenArsen — APPROVED** [review]

**Review by 64 — APPROVED** [review]

File: options/posix/include/sched.h [hunk 6]
```diff
@@ -73,6 +73,9 @@ int __mlibc_cpu_count(cpu_set_t *set);
 // Linux extension
 int clone(int (*)(void *), void *, int, void *, ...);
 
+// Glibc extension
+int sched_getcpu(void);
```
Comment (Geertiebear) [inline options/posix/include/sched.h:77]: This will now cause linker errors, this should be in a seperate header.

**Review by Geertiebear — CHANGES_REQUESTED** [review]

**Review by Dennisbonke — APPROVED** [review]
LGTM
